hathor/p2p/sync_v2/agent.py

```diff
@@ -545,36 +549,42 @@ def find_best_common_block(self,
         # Run an n-ary search in the interval [lo, hi).
         # `lo` is always a height where we are synced.
         # `hi` is always a height where sync state is unknown.
-        hi = min(peer_best_block.height, my_best_block.height)
-        lo = 0
-
-        lo_block_hash = self._settings.GENESIS_BLOCK_HASH
+        hi = min(peer_best_block, my_best_block)
+        lo = _HeightInfo(height=0, id=self._settings.GENESIS_BLOCK_HASH)
 
-        while hi - lo > 1:
+        while hi.height - lo.height > 1:
             self.log.info('find_best_common_block n-ary search query', lo=lo, hi=hi)
-            step = math.ceil((hi - lo) / 10)
-            heights = list(range(lo, hi, step))
-            heights.append(hi)
-
-            block_height_list = yield self.get_peer_block_hashes(heights)
-            block_height_list.sort(key=lambda x: x.height, reverse=True)
-
-            for height, block_hash in block_height_list:
+            step = math.ceil((hi.height - lo.height) / 10)
+            heights = list(range(lo.height, hi.height, step))
+            heights.append(hi.height)
+
+            block_info_list = yield self.get_peer_block_hashes(heights)
+            block_info_list.sort(key=lambda x: x.height, reverse=True)
+
+            # As we are supposed to be always synced at `lo`, we expect to receive a response
+            # with at least one item equals to lo. If it does not happen, we stop the search
+            # and return None. This might be caused when a reorg occurs during the search.
+            if not block_info_list:
+                return None
+            lo_block_info = block_info_list[-1]
+            if lo_block_info != lo:
+                return None
```

Comment:
**question**: Why -1?

Reply:
Because `lo` has the lowest height and `block_info_list` is sorted by decreasing height. So, if it belongs to the list, it will be at the last position.

Reply:
Ok, thanks!